src/awkward/operations/ak_from_avro_file.py

```diff
@@ -67,10 +67,13 @@ def from_avro_file(
 
 def _impl(form, length, container, highlevel, behavior):
 
-    return ak.from_buffers(
+    return ak.operations.ak_from_buffers._impl(
         form=form,
         length=length,
         container=container,
+        buffer_key="{form_key}-{attribute}",
+        backend="cpu",
         highlevel=highlevel,
         behavior=behavior,
+        simplify=True,
```

Comment:
This is internal code, so the name can be changed. (I suggested "reform" above.) The reason for "simplify" is because it calls "simplified" and a difference between these names would confuse developers. If `simplify=True`, then `ak_from_buffers._impl` internally calls `simplified`; if not, it does not.

The Avro implementation also makes non-canonical Forms, for the same reason as ArrayBuilder: putting an IndexedOptionArray around a UnionArray is the simplest and most symmetric thing to do, but it violates Arrow convention. Nobody cares what the Avro reader's internal Form was (same for ArrayBuilder), so changing it on the fly won't bother anybody.

Don't worry—this argument is absolutely hidden from users and we can redefine it at any time.